src/main.cpp

```diff
@@ -876,9 +982,21 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState &state, const C
                     return state.Invalid(false, REJECT_CONFLICT, "txn-mempool-conflict");
 
                 setConflicts.insert(ptxConflicting->GetHash());
+        COutPoint outpoint = tx.vin[i].prevout;
+        // A respend is a tx that conflicts with a member of the pool
+        if (pool.mapNextTx.count(outpoint))
+        {
+            fRespend = true;
+            // Relay only one tx per respent outpoint, but not if tx is equivalent to pool member
+            if (!doubleSpendFilter.contains(outpoint) && !tx.IsEquivalentTo(*pool.mapNextTx[outpoint].ptx))
+            {
+                relayForOutpoint = outpoint;
+                break;
             }
```

Comment:
Did this actually prevent double spends in XT?
Do we have any empirical results on this?